File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.js
function solution(board, moves) {
    let basket = [];
    var result = 0;
    for(move of moves) {
        for(row of board) {
            if (row[move-1] != 0) {
                let doll = row[move-1];
                row[move-1] = 0
                if(basket.length != 0 && basket[basket.length - 1] === doll) { 
                    result += 2
                    basket.pop()
                }
                else {
                    basket.push(doll)
                }
                break
            }
        }
    }

    return result;
}

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    box = []
    result = 0

    for move in moves:
        for row in board:
            if row[move - 1] != 0:
                doll = row[move - 1]
                row[move - 1] = 0
                if box and box[-1] == doll:
                    result += 2
                    box.pop()
                else:
                    box.append(doll)
                break

    return result
